src/main/java/seedu/address/logic/commands/AssignCommand.java

```diff
@@ -0,0 +1,253 @@
+                        nurse, nurseUidNo, patient);
+            }
+        } else {
+            sortAndCheckIndexForPatient(updatedDateSlotList);
+
+            for (Index index : dateslotIndex) {
+                DateSlot dateSlot = updatedDateSlotList.get(index.getZeroBased());
+                executeChecksAndActions(dateSlot, updatedHomeVisitList, updatedFullyScheduledList,
+                        nurse, nurseUidNo, patient);
+            }
+        }
+        editPatient(model, patient, updatedDateSlotList);
+        editNurse(model, nurse, updatedHomeVisitList, updatedFullyScheduledList);
+    }
+
+    public void checkInvalid(DateSlot dateSlot) throws CommandException {
+        if (dateSlot.getHasVisited() == true) {
+            throw new CommandException(String.format(MESSAGE_INVALID_DATESLOT, dateSlot.getDateSlotFormatted()));
+        }
+    }
+
+    public void checkAssigned(DateSlot dateSlot) throws CommandException {
+        if (dateSlot.getHasAssigned() == true) {
+            throw new CommandException(String.format(MESSAGE_ASSIGNED_DATESLOT, dateSlot.getDateSlotFormatted()));
+        }
+    }
+
+    public void checkCrashes(DateSlot dateSlot, List<HomeVisit> homeVisitList) throws CommandException {
+        Optional<HomeVisit> homeVisit = homeVisitList.stream().filter(
+                h -> h.getDateSlot().getDateTime().equals(dateSlot.getDateTime())).findFirst();
+        if (!homeVisit.isEmpty()) {
+            throw new CommandException(String.format(MESSAGE_TIME_CRASHES, dateSlot.getDateSlotFormatted()));
+        }
+    }
+
+    public void checkUnavailability(DateSlot dateSlot, Person nurse) throws CommandException {
+        List<Date> unavailabilityDateList = ((Nurse) nurse).getUnavailableDates();
+        Optional<Date> date = unavailabilityDateList.stream().filter(
+                d -> d.getDate().equals(dateSlot.getDate())).findFirst();
+        if (!date.isEmpty()) {
+            throw new CommandException(String.format(MESSAGE_UNAVAILABLE_DATE, dateSlot.getDateSlotFormatted()));
+        }
+    }
+
+    public void createHomeVisit(DateSlot date, Person patient, List<HomeVisit> homeVisitList,
+                                List<Date> updatedFullyScheduledDateList) {
+        HomeVisit homeVisit = new HomeVisit(date, ((Patient) patient).getUid().getUid());
+        homeVisitList.add(homeVisit);
+        LocalDate localdate = date.getDate();
+        int frequencyCount = 0;
+        for(int i = 0; i < homeVisitList.size(); i++) {
+            LocalDate dateToCheck = homeVisitList.get(i).getDateSlot().getDate();
+            if (dateToCheck.equals(localdate)) {
+                frequencyCount++;
+            }
+        }
+        if (frequencyCount == 4) {
+            updatedFullyScheduledDateList.add(new Date(localdate));
+        }
+    }
+
+    public void editPatient(Model model, Person patient, List<DateSlot> dateSlotList) throws CommandException {
```

Comment:
for defensive copy
and edit patient is like that cuz idw change the homevisit list here (if call edit command it will change homevisit list)

for edit nurse it will not affect anything so can just call editcommand